ngraph/core/src/layout.cpp

```diff
@@ -233,6 +233,33 @@ std::string Layout::to_string() const {
     return res.str();
 }
 
+namespace layout {
+
+std::vector<int64_t> find_permutation(const Layout& src_layout, const Rank& rank, const Layout& dst) {
```

Comment:
still don't why do we need `rank` :) but let's see how this implementation is changed once we support partial / dynamic layouts